questdb-rs/src/ingress/mod.rs

```diff
@@ -1011,6 +1069,175 @@ impl Buffer {
         Ok(self)
     }
 
+    /// Record a multidimensional array value for the given column.
+    ///
+    /// Supports arrays with up to [`MAX_ARRAY_DIMS`] dimensions. The array elements must
+    /// implement [`ArrayElement`] trait which provides type-to-[`ElemDataType`] mapping.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage with direct dimension specification:
+    ///
+    /// ```
+    /// # #[cfg(feature = "ndarray")]
+    /// # {
+    /// # use questdb::Result;
+    /// # use questdb::ingress::Buffer;
+    /// # use ndarray::array;
+    /// # fn main() -> Result<()> {
+    /// # let mut buffer = Buffer::new();
+    /// # buffer.table("x")?;
+    /// // Record a 2D array of f64 values
+    /// let array_2d = array![[1.1, 2.2], [3.3, 4.4]];
+    /// buffer.column_arr("array_col", &array_2d.view())?;
+    /// # Ok(())
+    /// # }
+    /// # }
+    ///
+    /// ```
+    ///
+    /// Using [`ColumnName`] for validated column names:
+    ///
+    /// ```
+    /// # #[cfg(feature = "ndarray")]
+    /// # {
+    /// # use questdb::Result;
+    /// # use questdb::ingress::{Buffer, ColumnName};
+    /// # use ndarray::Array3;
+    /// # fn main() -> Result<()> {
+    /// # let mut buffer = Buffer::new();
+    /// # buffer.table("x1")?;
+    /// // Record a 3D array of f64 values
+    /// let array_3d = Array3::from_elem((2, 3, 4), 42f64);
+    /// let col_name = ColumnName::new("col1")?;
+    /// buffer.column_arr(col_name, &array_3d.view())?;
+    /// # Ok(())
+    /// # }
+    /// # }
```

Comment:
I'd remove this example and show one with slices and another with vectors.

You can use some headings here.

Run
```
cargo doc --open
```

To preview.